appserver/transaction/jts/src/main/java/com/sun/jts/CosTransactions/GlobalTID.java

```diff
@@ -309,14 +308,16 @@ public final int hashCode() {
         hashCode = 0;
 
         // Add up the values in the XID.
-
-        if( realTID.tid != null )
-            for( int pos = 0; pos < realTID.tid.length; pos++ )
+        // GLASSFISH-21532 The branch qualifier values should not be used for calculating the hashcode
+        if( realTID.tid != null ) {
+            for( int pos = 0; pos < realTID.tid.length - realTID.bqual_length; pos++ ) {
                 hashCode += realTID.tid[pos];
+            }
+        }
 
         // Add in the formatId and branch qualifier length.
```

Comment:
The most minor of complaints: can you fix this comment please? It no longer adds the bqual length